src/components/Connection.tsx

```diff
@@ -505,21 +533,63 @@ const Connection: React.FC<ConnectionProps> = ({
             </Tooltip>
           </TooltipProvider>
         )}
-        {datasets.length > 0 && (
+       {isConnected && datasets.length > 0 && (
           <TooltipProvider>
             <Tooltip>
-              <Button onClick={saveData}>
-                <TooltipTrigger asChild>
-                  {datasets.length === 1 ? (
-                    <FileDown />
-                  ) : (
-                    <span className="flex flex-row justify-center items-center">
-                      <FileArchive />
-                      <p className=" text-lg">{`(${datasets.length})`}</p>
-                    </span>
-                  )}
-                </TooltipTrigger>
-              </Button>
+              <div className="flex">
+                <Button onClick={saveData} className="rounded-r-none">
+                  <TooltipTrigger asChild>
+                    {datasets.length === 1 ? (
+                      <FileDown className="mr-2" />
+                    ) : (
+                      <span className="flex flex-row justify-center items-center">
+                        <FileArchive className="mr-2" />
+                        <p className="text-lg">{datasets.length}</p>
+                      </span>
+                    )}
+                  </TooltipTrigger>
+                </Button>
+                <Separator orientation="vertical" className="h-full" />
+                {datasets.length === 1 ? (
+                  <Button className="rounded-l-none" onClick={deletedata}>
+                    <Trash2 size={20} />
+                  </Button>
+                ) : (
+                  <>
+                  <Popover open={open} onOpenChange={setOpen}>
+                    <PopoverTrigger asChild>
+                      <Button className="rounded-none">
+                        <ArrowUp size={20} />
+                      </Button>
+                    </PopoverTrigger>
+                    <Button className="rounded-l-none" onClick={deletedata}>
+                    <Trash2 size={20} />
+                  </Button>
+                    <PopoverContent className="w-80">
+                      <div className="space-y-4">
+                        {datasets.map((dataset, index) => (
+                          <div
+                            key={index}
+                            className="flex justify-between items-center"
+                          >
+                            <span>file{index}.csv</span>
```

Comment:
We are saving files with name data1.csv data2.csv and so on, not file1 and file2, so change one of any. They should match.